components/Input/Input.tsx

```diff
@@ -0,0 +1,59 @@
+import { forwardRef } from 'react';
+import { VariantProps } from 'class-variance-authority';
+import { FieldError, useFormContext } from 'react-hook-form';
+import InputClasses from './Input.classes';
+
+interface InputBaseProps extends VariantProps<typeof InputClasses> {
+  label?: string;
+  name: string;
+}
+
+// TODO: Icon will be displayed in error situations
+type Props = React.InputHTMLAttributes<HTMLInputElement> & InputBaseProps;
+
+/**
+ * @example <Input name="email" label="Email" placeholder="olivia@untitledui.com"  />
```

Comment:
we can remove this example from here. we will create the component documents later.